deeppixel/image_stitching_2/code/pano.py

```diff
@@ -0,0 +1,141 @@
+import numpy as np
+import cv2
+import sys
+from matchers import matchers
+
+class Stitch:
+	def __init__(self, args):
+		self.path = args
+		fp = open(self.path, 'r')
+		filenames = [each.rstrip('\r\n') for each in  fp.readlines()]
+		#print filenames
+		self.images = [cv2.resize(cv2.imread(each),(480, 320)) for each in filenames]
+		self.count = len(self.images)
+		self.left_list, self.right_list, self.center_im = [], [],None
+		self.matcher_obj = matchers()
+		self.prepare_lists()
+
+	def prepare_lists(self):
+		#print "Number of images : %d"%self.count
+		self.centerIdx = self.count/2 
+		#print "Center index image : %d"%self.centerIdx"""
+		self.center_im = self.images[int(self.centerIdx)]
+		for i in range(self.count):
+			if(i<=self.centerIdx):
+				self.left_list.append(self.images[i])
+			else:
+				self.right_list.append(self.images[i])
+		#print "Image lists prepared"
+
+	def leftshift(self):
+		# self.left_list = reversed(self.left_list)
+		a = self.left_list[0]
+		for b in self.left_list[1:]:
+			H = self.matcher_obj.match(a, b, 'left')
+			#print "Homography is : ", H
+			xh = np.linalg.inv(H)
+			#print "Inverse Homography :", xh
+			ds = np.dot(xh, np.array([a.shape[1], a.shape[0], 1]));
+			ds = ds/ds[-1]
+			#print "final ds=>", ds
+			f1 = np.dot(xh, np.array([0,0,1]))
+			f1 = f1/f1[-1]
+			xh[0][-1] += abs(f1[0])
+			xh[1][-1] += abs(f1[1])
+			ds = np.dot(xh, np.array([a.shape[1], a.shape[0], 1]))
+			offsety = abs(int(f1[1]))
+			offsetx = abs(int(f1[0]))
+			dsize = (int(ds[0])+offsetx, int(ds[1]) + offsety)
+			#print "image dsize =>", dsize
+			tmp = cv2.warpPerspective(a, xh, dsize)
+			# cv2.imshow("warped", tmp)
+			# cv2.waitKey()
+			tmp[offsety:b.shape[0]+offsety, offsetx:b.shape[1]+offsetx] = b
```

Comment:
I guess we don't need a keyboard interrupt? Remove any unrelated commented out code